src/egl/drivers/dri2/platform_android.c

```diff
@@ -1137,19 +1173,32 @@ dri2_initialize_android(_EGLDriver *drv, _EGLDisplay *dpy)
    dri2_dpy = calloc(1, sizeof(*dri2_dpy));
    if (!dri2_dpy)
       return _eglError(EGL_BAD_ALLOC, "eglInitialize");
-
-   ret = hw_get_module(GRALLOC_HARDWARE_MODULE_ID,
                        (const hw_module_t **)&dri2_dpy->gralloc);
-   if (ret) {
-      err = "DRI2: failed to get gralloc module";
+   
+   ret = update_pfn(dri2_dpy);
```

Comment:
this should be called only if we have gralloc1 in use and maybe called resolve_gralloc1_functions()